template/nuxt/pages/index.vue

```diff
@@ -28,6 +28,18 @@
   </div>
 </template>
 
+<%_ if (typescript) { _%>
+<script lang="ts">
+import Vue from 'vue'
+import Logo from '~/components/Logo.vue'
+
+export default Vue.extend({
+  components: {
+    Logo
```

Comment:
I think we could update both TS/JS (if/else) snippets to make it use lazy loading of component to kind of directly show case a best practice to new users ?
```suggestion
    Logo: () => import('~/components/Logo.vue')
```

Reply:
Still => May be not relevant here, it's much relevant when there are `v-if` of nested routing :)

Reply:
Let's think about it outside of this PR context